packages/wallet/src/cip30.ts

```diff
@@ -49,9 +50,23 @@ export type SubmitTxCallbackParams = {
   data: Cardano.Tx;
 };
 
-export type CallbackConfirmation = (
-  args: SignDataCallbackParams | SignTxCallbackParams | SubmitTxCallbackParams
-) => Promise<boolean>;
+// Optional callback
+export type GetCollateralCallbackParams = {
+  type: Cip30ConfirmationCallbackType.GetCollateral;
+  data: {
+    amount: string;
```

Comment:
Our types/interfaces should be operating with core types rather than serialized
```suggestion
    amount: Cardano.Lovelace;
```